prov/rxm/src/rxm_cq.c

```diff
@@ -374,18 +384,18 @@ ssize_t rxm_cq_handle_data(struct rxm_rx_buf *rx_buf)
 					fi_mr_desc(rx_buf->recv_entry->rxm_iov.desc[i]);
 		}
 
-		ret = rxm_match_rma_iov(rx_buf->recv_entry, rx_buf->rma_iov,
-					rx_buf->match_iov);
-		if (ret)
-			return ret;
-
+		/* Ignore the case when the posted recv buffer is not large enough,
+		 * FI_TRUNC error will be generated to user at the end */
+		rxm_match_rma_iov(rx_buf->recv_entry, rx_buf->rma_iov, rx_buf->match_iov);
 		RXM_LOG_STATE_RX(FI_LOG_CQ, rx_buf, RXM_LMT_READ);
 		rx_buf->hdr.state = RXM_LMT_READ;
 		return rxm_lmt_rma_read(rx_buf);
 	} else {
-		ofi_copy_to_iov(rx_buf->recv_entry->rxm_iov.iov, rx_buf->recv_entry->rxm_iov.count, 0,
-				rx_buf->pkt.data, rx_buf->pkt.hdr.size);
-		return rxm_finish_recv(rx_buf);
+		return rxm_finish_recv(rx_buf,
+				       ofi_copy_to_iov(rx_buf->recv_entry->rxm_iov.iov,
+						       rx_buf->recv_entry->rxm_iov.count,
+						       0, rx_buf->pkt.data,
+						       rx_buf->pkt.hdr.size));
```

Comment:
nit: This would look better if you saved the result of ofi_copy_to_iov() into a variable and passed that to rxm_finish_recv().

Reply:
Sure. I've applied this comment